conf/application.conf

```diff
@@ -17,21 +17,40 @@ play {
       maxMemoryBuffer = 250M
       maxDiskBuffer = 1G
     }
-    filters = "com.scalableminds.webknossos.datastore.Filters"
     fileMimeTypes = ${play.http.fileMimeTypes} """
       wasm=application/wasm
     """
   }
-  filters.headers {
-    # Production Content Security Policy
-    # The sha256 hash in the script-src refers to the google analytics configuration script in main.scala.html.
-    # contentSecurityPolicy = "default-src 'self'; script-src 'self' 'unsafe-eval' *.googletagmanager.com app.olvy.co webknossos.olvy.co *.gstatic.com 'sha256-NA81o2FV8jh0UizSPsBz2qEGpXfap54Eghuo7vxrH8g='; style-src 'self' 'unsafe-inline' *.gstatic.com; connect-src 'self' app.olvy.co api.github.com gist.github.com *.google-analytics.com api.airbrake.io; frame-src webknossos.olvy.co; img-src 'self' data:"
-    # Development Content Security Policy
-    # In addition to the production policies, the development policy contains the script-src 'unsafe-inline' (to make the React Dev Tools work in Firefox) and the connect-src 'ws://localhost:9002' (webpack dev-server).
-    contentSecurityPolicy = "default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' *.googletagmanager.com  app.olvy.co webknossos.olvy.co *.gstatic.com; style-src 'self' 'unsafe-inline' *.gstatic.com; connect-src 'self' ws://localhost:9002 app.olvy.co api.github.com gist.github.com *.google-analytics.com api.airbrake.io; frame-src webknossos.olvy.co; img-src 'self' data:"
-    contentTypeOptions = null
-    frameOptions = null
-    xssProtection = null # Not supported on modern browsers
+  filters {
+    # Note that the CSP filter is not enabled. We manually add this header ourselves for the
+    # main view, using the following config (note that currently only directives are used, not nonce or hashes)
```

Comment:
> not nonce or hashes

But line 32 contains a hash?! This way of setting the CSP is simply more structured as it breaks the CSP into is directive parts that writing everything at once. Or am I misunderstanding something? :thinking:

Reply:
There is a dedicated setting for hashes https://www.playframework.com/documentation/2.8.x/CspFilter#Configuring-CSP-Hashes which we don’t use here. However, it looks like only using the `directives` setting results in the header being set as we wanted, including the hash in line 32.